src/apps/concentrator/ethereum/concentrator.balance-fetcher.ts

```diff
@@ -0,0 +1,73 @@
+import { Inject } from '@nestjs/common';
+
+import { drillBalance } from '~app-toolkit';
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { Register } from '~app-toolkit/decorators';
+import { presentBalanceFetcherResponse } from '~app-toolkit/helpers/presentation/balance-fetcher-response.present';
+import { BalanceFetcher } from '~balance/balance-fetcher.interface';
+import { isSupplied } from '~position/position.utils';
+import { Network } from '~types/network.interface';
+
+import { CONCENTRATOR_DEFINITION } from '../concentrator.definition';
+import { ConcentratorContractFactory } from '../contracts';
+
+const network = Network.ETHEREUM_MAINNET;
+
+@Register.BalanceFetcher(CONCENTRATOR_DEFINITION.id, network)
+export class EthereumConcentratorBalanceFetcher implements BalanceFetcher {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(ConcentratorContractFactory) private readonly concentratorContractFactory: ConcentratorContractFactory,
+  ) {}
+
+  async getTokenBalances(address: string) {
+    return await this.appToolkit.helpers.tokenBalanceHelper.getTokenBalances({
+      address,
+      appId: CONCENTRATOR_DEFINITION.id,
+      groupId: CONCENTRATOR_DEFINITION.groups.acrv.id,
+      network: Network.ETHEREUM_MAINNET,
+    });
+  }
+
+  async getPoolBalances(address: string) {
+    return this.appToolkit.helpers.contractPositionBalanceHelper.getContractPositionBalances({
```

Comment:
You might find the `masterChefContractPositionBalanceHelper` helper more useful here.